packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -242,6 +242,15 @@ class ResidentWebRunner extends ResidentRunner {
       );
     } on SocketException catch (err) {
       throwToolExit(err.toString());
+    } on StateError catch (err) {
+      final String message = err.toString();
+      if (message.contains('Unable to start build daemon')) {
```

Comment:
Seems a bit brittle? What if the message changes? Should we have a test that triggers this error on purpose and verifies the message contents?

Reply:
Ideally we wouldn't have to catch this at all. cc @jakemac53, though it might be related to use of the builders key. If we regress here, then we're only as bad as we were before though

Reply:
We could throw a special exception if that would help - it looks like just a `StateError` today.

Reply:
That would probably be ideal. I'm going to land this as is, but if this is updated to a specific error type I could catch that instead.